### ISSUE TITLE
Known issues with JSX and eslint-plugin-react 

### ISSUE THREAD

**Issue by soda0289** [eslint/typescript-eslint-parser#213]
Since typescript v2.2 and typescript-eslint-parser v2.1 JSX parsing has significantly improved. All of the eslint rules should work as expected with JSX. We have also tested the eslint-plugin-jsx-a11y and have found no issues.

Most of the rules in eslint-plugin-react work except for the following:
```
react/no-unescaped-entities
react/self-closing-comp
react/void-dom-elements-no-children
react/jsx-indent
react/jsx-indent-props
react/jsx-max-props-per-line
```

Work still needs to be done to investigate the issue with each rule to determine exactly where the problem comes from.

**Comment by corbinu** [comment]
@soda0289 I will make sure to add the react plugin to my smoke tests. I know I still need to release a new version. Just been busy with work work :)

**Comment by soda0289** [comment]
Thanks @corbinu. We will need to find some projects that use typescript and react.
The only one I can think of is [Apollo React Client](https://github.com/apollographql/react-apollo) and it is tiny. We should start working on integration tests between all these rules and typescript parser as well.

I don't think the rules cause any crashes just false positives or incorrect line/column numbers. I noticed that the jsx-indent rules will cause eslint to run forever but have not tried to reproduce on a code snippet.

We should also make a list of known issues for eslint core rules and popular plugins too as I think it will help reduce the amount of issues for this project.

**Comment by corbinu** [comment]
I just added the plugin and https://github.com/alm-tools/alm

Will try to push the update tomorrow and then shouldn't be hard for us to make a list.

**Comment by sompylasar** [comment]
Refs https://github.com/eslint/typescript-eslint-parser/issues/266#issuecomment-302839595 re: `react/no-unescaped-entities`.

**Comment by soda0289** [comment]
@sompylasar https://github.com/eslint/typescript-eslint-parser/pull/277 fixes `no-unescaped-entites`

**Comment by soda0289** [comment]
These rules are now fixed in the master branch and will be part of the next release

**Comment by sompylasar** [comment]
Thanks @soda0289 !

**Comment by OliverJAsh** [comment]
Would it be possible to push a release out now?

**Comment by JamesHenry** [comment]
@OliverJAsh we have just cut a release for the latest work https://github.com/eslint/typescript-eslint-parser/releases/tag/v4.0.0

**Comment by sarod** [comment]
README.MD still references this issue in the known issues list. Is it on purpose?

**Comment by soda0289** [comment]
@sarod The README can be updated. Haven't since any other issues reported from the react plugin

**Comment by OliverJAsh** [comment]
These rules now work, however I think there are issues with the auto fixes—they seem to be corrupting the code. See https://github.com/eslint/typescript-eslint-parser/issues/343.
